src/main/webapp/src/components/youTubeRecommendations.js

```diff
@@ -0,0 +1,110 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { withStyles } from "@material-ui/core/styles";
+import YouTube from "react-youtube";
+import List from "@material-ui/core/List";
+import ListItem from "@material-ui/core/ListItem";
+import CircularProgress from '@material-ui/core/CircularProgress';
+import axios from "axios";
+
+const styles = (theme) => ({
+  youTubeVideo: {
+    playerVars: {
+      autoplay: 1,
+    },
+  },
+  youTubeRecommendationsList: {
+    padding: 0,
+    display: "flex",
+    flexDirection: "row",
+    justifyContent: "center",
+    flexWrap: "wrap",
+  },
+  youTubeRecommendationsListItem: {
+    width: "640px",
+    margin: "10px",
+    listStyleType: "none",
+  },
+});
+
+/**
+ * Displays sentiment analysis info for the lyrics.
+ */
+class YouTubeRecommendations extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      videoIds: [],
+      isLoading: false,
+      error: null,
+    };
+  }
+
+  componentDidMount() {
+    this.setState({ isLoading: true });
+
+    axios
+      .get(`/youtube?query=${this.props.q}`)
+      .then((result) => result.data)
+      .then((videoIds) => {
+        this.setState({
+          videoIds: videoIds,
+          isLoading: false,
+        })
+      })
+      .catch((error) =>
+        this.setState({
+          error,
+          isLoading: false,
+        })
+      );
+  }
+
+  render() {
+    const classes = this.props.classes;
+
+    if (this.state.error) {
+      return (
+        <div>
+          <p>{this.state.error.message}</p>
```

Comment:
Again, the user might not know what this error means.

Reply:
Usually it's "Internal Server error 500", I think there's no need to explain the user why it happened, because they shouldn't care about it)

Reply:
Well, if the user shouldn't care about it, then it shouldn't be in the UI in the first place. I think it's a better idea to give the user easier to process information. Maybe change it to `Oops, something went wrong` as in the other request?

Reply:
Yep